aggregate_datasets.py

```diff
@@ -40,13 +41,247 @@ def parse_args():
+        },
+        "type": {
+          "dtype": "string",
+          "id": None,
+          "_type": "Value"
+        },
+        "value": {
+          "dtype": "string",
+          "id": None,
+          "_type": "Value"
+        }
+      }
+    ],
+    "seed_id": {
+      "dtype": "int32",
+      "id": None,
+      "_type": "Value"
+    },
+    "text": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url_host_name": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url_host_registered_domain": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url_host_tld": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url_surtkey": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "warc_filename": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "warc_record_length": {
+      "dtype": "int32",
+      "id": None,
+      "_type": "Value"
+    },
+    "warc_record_offset": {
+      "dtype": "int32",
+      "id": None,
+      "_type": "Value"
+    }
+  }
+    return Features(convert_types(features))
+
+def collapse_meta_(batch):
+    """{"text": str, "meta": str}"""
+    # TODO: check that
+    columns_not_in_meta = ["text", "html_str"]
+    columns_to_collapse = [name for name in batch.keys() if name not in columns_not_in_meta]
+
+    new_batch = {
+        "text": batch["text"],
+        "meta": [
+            str({key: value for key, value in zip(columns_to_collapse, row)})
+            for row in zip(*[batch[name] for name in columns_to_collapse])
+        ]
+    }
+    return new_batch
+
+def collapse_meta(ds: Dataset, num_proc: int):
+    """{"text": str, "meta": str}"""
+    columns_to_keep = ["text"]
+    column_names_to_remove = [name for name in ds.column_names if name not in columns_to_keep]
+    return ds.map(collapse_meta_, batched=True, num_proc=num_proc, remove_columns=column_names_to_remove)
 
 def load_datasets(args):
-    ds_name, ratio, split, seed = args
-    ds = load_dataset(ds_name, use_auth_token=os.environ["HF_USER_ACCESS_TOKEN"], ignore_verifications=True)
+    ds_ratio, split, seed = args
+    ds_name = ds_ratio["dataset_path"]
+    ratio = ds_ratio["ratio"]
+    if ds_ratio["is_catalogue"]:
+        ds = load_dataset(ds_name, use_auth_token=os.environ["HF_USER_ACCESS_TOKEN"], ignore_verifications=True)
+    else:
+        # We assume it comes from pseudo crawl.
+        # Pseudo crawl needs to be downloaded locally beforehand.
+        features = get_features()
+        dataset_path = Path(ds_name)
+        ds = load_dataset(str((dataset_path / "text__html").absolute()), data_files="**.jsonl.gz", features=features)
+        # collapse all meta data in "meta" column
+        ds = collapse_meta(ds, num_proc=1)
```

Comment:
Once you add the field "source_dataset", maybe you could merge master branch into this branch and then move this line
```
# collapse all meta data in "meta" column
ds = collapse_meta(ds, num_proc=1)
```
to the `# Process meta:` section, again with an if for catalogue vs pseudo-crawl.